pkg/controllers/leaderworkerset_controller.go

```diff
@@ -523,8 +558,54 @@ func (r *LeaderWorkerSetReconciler) iterateReplicas(ctx context.Context, lws *le
 	return continuousReadyReplicas, lwsUnreadyReplicas, nil
 }
 
+func (r *LeaderWorkerSetReconciler) getLeaderStatefulSet(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet) (*appsv1.StatefulSet, error) {
+	sts := &appsv1.StatefulSet{}
+	err := r.Get(ctx, types.NamespacedName{Name: lws.Name, Namespace: lws.Namespace}, sts)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return sts, nil
+}
+
+func (r *LeaderWorkerSetReconciler) getOrCreateRevisionIfNonExist(ctx context.Context, sts *appsv1.StatefulSet, lws *leaderworkerset.LeaderWorkerSet) (*appsv1.ControllerRevision, error) {
+	revisionKey := ""
+	if sts != nil {
+		// Uses the hash in the leader sts to avoid detecting update in the case where LWS controller is upgraded from a version where
+		// the revisionKey was used to detect update instead of controller revision.
+		revisionKey = revisionutils.GetRevisionKey(sts)
+	}
```

Comment:
```suggestion
	// Uses the hash in the leader sts to avoid detecting update in the case where LWS controller is upgraded from a version where
	// the revisionKey was used to detect update instead of controller revision.
	revisionKey := revisionutils.GetRevisionKey(sts)
```

Reply:
GetRevisionKey already does a nil check